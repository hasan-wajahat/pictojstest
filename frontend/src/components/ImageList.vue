<template>
  <div class="image">
    <h4>Images</h4>
    <ul class="list-unstyled">
      <li
        :key="index"
        v-for="(image, index) in images"
        class="image-asset"
        @click="$emit('asset-click', {url: image, type: 'image'})"
      >
        <img :src="image" class="img-rounded non-draggable">
      </li>
    </ul>
  </div>
</template>


<script>
export default {
  name: "ImageList",
  props: {
    images: Array
  }
};
</script>

<style lang="scss" scoped>
.image-asset {
  cursor: pointer;
  transition: transform 0.2s;
  &:hover {
    transform: scale(1.1);
  }
}
.non-draggable {
  user-drag: none;
  user-select: none;
  -moz-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
  -ms-user-select: none;
}
</style>