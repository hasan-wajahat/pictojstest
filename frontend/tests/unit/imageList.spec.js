import { shallowMount } from '@vue/test-utils'
import ImageList from '@/components/ImageList.vue'

const IMAGES = ['first', 'second'];

describe('ImageList.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ImageList, {
      propsData: {
        images: IMAGES
      }
    });
  })

  it('renders correct number of images', () => {
    expect(wrapper.findAll('img').length).toEqual(IMAGES.length);
  })

  it('passes correct source to image', () => {
    expect(wrapper.find('img').attributes('src')).toBe(IMAGES[0]);
  })
  
  it('button emits proper event', () => {
    wrapper.find('li').trigger('click');
    expect(wrapper.emitted()).toMatchSnapshot();
  })
})
