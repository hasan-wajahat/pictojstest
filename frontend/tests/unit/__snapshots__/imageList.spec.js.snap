// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList.vue button emits proper event 1`] = `
Object {
  "asset-click": Array [
    Array [
      Object {
        "type": "image",
        "url": "first",
      },
    ],
  ],
}
`;
